smart-home-provider/cloud/auth-provider.js

```diff
@@ -304,6 +302,53 @@ Auth.registerAuth = function(app) {
 // &grant_type=authorization_code
 
 
+/**
+ * Decode authorization into client_id and client_secret.
+ * @return {{}}
+ * {
+ *   client_id: "CLIENT_ID",
+ *   client_secret: "CLIENT_SECRET"
+ * }
+ */
+function getHeaderAuthorization(req){
+  let headerAuthor = {};
+    try{
+      if(!!req.headers.authorization){
+        let authorizationString = Buffer.from(req.headers.authorization.split(" ")[1], 'base64').toString();
+        if(!!authorizationString){
+          let authorizationArray = authorizationString.split(':');
```

Comment:
Could we destructure this directly, making the object creation simpler, ie.

```
const [client_id, client_secret] = authorizationString.split(':')
return {
  client_id,
  client_secret
}
```

Reply:
Oh, I forgot that we can do that nowadays!
OK, I'll do it.